src/operand/reference.ts

```diff
@@ -27,14 +37,29 @@ function contextValueLookup (ctx: Context, key: string): Result {
   let pointer = ctx
 
   for (let i = 0; i < keys.length; i++) {
+    let currentValue = pointer[keys[i]]
+
+    // Resolve array notation
+    const matchesArrayNotation = /^(.+)\[(\d+)]$/.exec(keys[i])
```

Comment:
to be consistent, escape the "]" as well, i.e. ```\]```

Reply:
13b35e60462b79c6125c239e32eb7acd8406cdba